enterprise/backend/src/metabase_enterprise/workspaces/mirroring.clj

```diff
@@ -1,48 +1,53 @@
+                            new-s+n   (table-mapping old-s+n old-s+n)]
+                        (assert (= "table" (:type target)) "We only support mirroring transforms that target tables.")
+                        (assert (= database-id (:database target)) "Unexpected target database for transform.")
+                        (merge transform
+                               ;; TODO (Chris 2025-11-19) shouldn't rename like this - will mess up promotion!
+                               {:name         (str (:name transform) "_DUP")
+                                :workspace_id (:id workspace)
+                                ;; TODO (Chris 2025-11-19) remap source table (when necessary)
+                                #_#_:source {}
+                                :target       {:type     "table"
+                                               :database database-id
+                                               :schema   (first new-s+n)
+                                               :name     (second new-s+n)}}))
+        mirror        (t2/insert-returning-instance! :model/Transform mirror)
+        graph-node    (assoc transform :mapping (select-keys mirror [:id :name]))
+        _             (t2/insert! :model/WorkspaceMappingTransform
+                                  {:upstream_id   (:id transform)
+                                   :downstream_id (:id mirror)
+                                   :workspace_id  (:id workspace)})]
+    (update graph :transforms #(conj (or % []) graph-node))))
 
-        mirror        (t2/insert-returning-instance!
-                       :model/Transform
-                       ;; TODO (Chris 2025-11-19) avoid reselecting transform here, by passing along all the keys from the top
-                       (merge (-> (t2/select-one (into [:model/Transform] t2-transform-mirrored-keys) (:id transform))
-                                  ;; TODO (Chris 2025-11-19) shouldn't rename like this - will mess up promotion!
-                                  ;; this should be happening elsewhere, not on write, for now ok
-                                  (update :name str "_DUP")
-                                  (assoc :workspace_id (:id workspace)))
-                              ;; TODO (Chris 2025-11-19) remap source table (when necessary)
-                              {:target {:type     "table"
-                                        :database database-id
-                                        :schema   (:schema output-table)
-                                        :name     (:name output-table)}}))]
-    (t2/insert! :model/WorkspaceMappingTransform
-                {:upstream_id   (:id transform)
-                 :downstream_id (:id mirror)
-                 :workspace_id  (:id workspace)})
-    (assoc graph :transforms [(assoc transform :mapping (select-keys mirror [:id :name]))])))
+(defn- mirror-transforms! [workspace database-id graph]
```

Comment:
The narrower the scope, the less this guidance applies. I highly doubt anyone could get confused here!